feat(client): add a select component for predefined engine types by Mayank Agarwal

22mayank opened the pull request.

mayankagarwal2 committed:
feat(client): add a select component for predefined engine types

neelneelneel commented on packages/client/src/components/block-defaults/select-engine-block/config.tsx:
```diff
@@ -0,0 +1,118 @@
+    widget: 'select-engine',
+    type: BLOCK_TYPE_INPUT,
+    data: {
+        style: {
+            padding: '4px',
+        },
+        value: '',
+        label: 'Example Select Engine Input',
+        hint: '',
+        options: [],
+        required: false,
+        disabled: false,
+        loading: false,
+        optionLabel: '',
+        optionSublabel: '',
+        optionValue: '',
+        multiple: false,
```
Remove extra / unnecessary attributes

neelneelneel commented on packages/client/src/components/block-defaults/select-engine-block/config.tsx:
```diff
@@ -0,0 +1,118 @@
+    },
+    render: SelectEngineBlock,
+    icon: ViewList,
+    contentMenu: [
```
Remove unnecessary settings

neelneelneel commented on packages/client/src/components/block-defaults/select-engine-block/SelectEngineBlock.tsx:
```diff
@@ -0,0 +1,167 @@
+
+export interface SelectEngineBlockDef extends BlockDef<'select-engine'> {
+    widget: 'select-engine';
+    data: {
```
Add a filter for the various catalog types.

neelneelneel reviewed (changes requested):
Remvo

mayankagarwal2 committed:
Merge branch 'dev' into feature/add-select-engine-dropdown-component

mayankagarwal2 committed:
feat(client): addressed PR feedback

22mayank commented:
Hi Neel, I have addressed the PR feedback for removing unused attributes & unnecessary settings. The filter for various catalog is already present. ( Attached screenshot )
![image](https://github.com/user-attachments/assets/dec4ca20-033a-4d91-8c04-c793c01ed72e)

neelneelneel reviewed (changes requested):
Few changes
- [ ] Move the call to get engines to `SelectEngineBlock.tsx`. This will enable infinite-loading
- [ ] Options can be removed, so Data should be defined as
```
export interface SelectEngineBlockDef extends BlockDef<'select-engine'> {
    widget: 'select-engine';
    data: {
        style: CSSProperties;
        label: string;
        type: string // corresponds to engine type
        value: {name:string; id:string} // corresponds to name + id
        required: boolean;
        disabled: boolean;
        hint?: string;
        loading?: boolean;
    };
}
```

neelneelneel commented:
@themaherkhalil  - how should this interact with dependencies?

neelneelneel commented:
Per discussion, doesn't interact with dependencies as it is user-dependent

johbaxter commented:
Not a viable approach
